### ISSUE TITLE
[tool_crash] NoSuchMethodError: The getter 'host' was called on null.Receiver in package:devtools_server ClientManager._areSameVmServices

### ISSUE THREAD

**Issue by AdamTV** [flutter/flutter#72747]
## Command
```
flutter run -d chrome --web-hostname localhost --web-port 5000
```

## Steps to Reproduce
1. Run above command
2. Launch Dart DevTools
3. Connect to Flutter app
4. Attempt to launch DevTools again

## Logs
NoSuchMethodError: NoSuchMethodError: The getter 'host' was called on null.
Receiver: null
Tried calling: host
```
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      ClientManager._areSameVmServices (package:devtools_server/src/client_manager.dart:62:30)
#2      ClientManager.findExistingConnectedClient.<anonymous closure> (package:devtools_server/src/client_manager.dart:57:32)
#3      ListMixin.firstWhere (dart:collection/list.dart:161:15)
#4      ClientManager.findExistingConnectedClient (package:devtools_server/src/client_manager.dart:55:21)
#5      _tryReuseExistingDevToolsInstance (package:devtools_server/src/server.dart:719:34)
#6      launchDevTools (package:devtools_server/src/server.dart:828:13)
#7      DevtoolsServerLauncher.launch (package:flutter_tools/src/isolated/devtools_launcher.dart:29:13)
<asynchronous suspension>
#8      ResidentRunner.launchDevTools (package:flutter_tools/src/resident_runner.dart:1277:5)
<asynchronous suspension>
#9      TerminalHandler.processTerminalInput (package:flutter_tools/src/resident_runner.dart:1602:7)
<asynchronous suspension>
```
```
[✓] Flutter (Channel beta, 1.24.0-10.2.pre, on Microsoft Windows [Version 10.0.19041.685], locale en-CA)
    • Flutter version 1.24.0-10.2.pre at C:\Users\adams\source\repos\flutter
    • Framework revision 022b333a08 (5 weeks ago), 2020-11-18 11:35:09 -0800
    • Engine revision 07c1eed46b
    • Dart version 2.12.0 (build 2.12.0-29.10.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\adams\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] IntelliJ IDEA Community Edition (version 2020.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.2.2
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.51.1)
    • VS Code at C:\Users\adams\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.17.0

[✓] Connected device (3 available)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 87.0.4280.88
    • Edge (web)       • edge       • web-javascript • Microsoft Edge 87.0.664.66

• No issues found!

```

## Flutter Application Metadata
**Type**: app
**Version**: null
**Material**: true
**Android X**: false
**Module**: false
**Plugin**: false
**Android package**: null
**iOS bundle identifier**: null
**Creation channel**: beta
**Creation framework version**: 022b333a089afb81c471ec43d1f1f4f26305d876
### Plugins
cloud_firestore-0.14.4
cloud_firestore_web-0.2.1+2
firebase_auth-0.18.4+1
firebase_auth_web-0.3.2+3
firebase_core-0.5.3
firebase_core_web-0.2.1+1
google_sign_in-4.5.6
google_sign_in_web-0.9.2

**Comment by jmagman** [comment]
\cc @DanTup @kenzieschmoll https://github.com/flutter/devtools/pull/2489

**Comment by DanTup** [comment]
Based on the error and call stack, I think it may be trying to launch DevTools with a `null` VM Service URL, perhaps from here:

```
#6      launchDevTools (package:devtools_server/src/server.dart:828:13)
#7      DevtoolsServerLauncher.launch (package:flutter_tools/src/isolated/devtools_launcher.dart:29:13)
```

However, if I understand https://github.com/flutter/flutter/pull/71737 correctly, it looks like that code may have been removed and DevTools is now spawned as a separate process rather than being called in-process so that may have affected (or even fixed) that (@kenzieschmoll?).

**Comment by kenzieschmoll** [comment]
That sounds right, Danny. The stack trace looks outdated.

**Comment by jmagman** [comment]
@kenzieschmoll @DanTup can we close this then as fixed in master?

**Comment by kenzieschmoll** [comment]
@AdamTV if you upgrade your flutter to latest, can you reproduce this issue?

I tried to reproduce myself but have some questions on your repro steps.

> Launch Dart DevTools

How did you launch DevTools? pressing 'v'? Launching from pub?

> Connect to Flutter app

Connect to the web app you just launched?

> Attempt to launch DevTools again

Did you close the first instance of DevTools? How are you launching for this second launch of DevTools?

**Comment by kenzieschmoll** [comment]
Let's close this anyway - the stack trace is from code that no longer exists.

**Comment by AdamTV** [comment]
@kenzieschmoll I can confirm after upgrading flutter to beta version (1.25.0-8.1.pre), I cannot reproduce this issue! To answer your questions:

> How did you launch DevTools? pressing 'v'? Launching from pub?
I used the 'Open DevTools' icon in Android Studio

> Connect to the web app you just launched?
That is correct indeed.

> Did you close the first instance of DevTools? How are you launching for this second launch of DevTools?
I did not close the first instance of DevTools. Launched second instance of DevTools same way as first.

In my attempt to recreate this issue I followed my steps in initial answer using details described in this reply, still unable to recreate. Flutter FTW!
